### ISSUE TITLE
when changing orientation the aspect ratio is degraded

### ISSUE THREAD

**Issue by joaofcosta-pt** [maxkonovalov/MKRingProgressView#39]
Hi there,

 I have a round MKRingProgressViewGroup and whenever I rotate my device the individual rings appear distorted until their progress is affected. To perform a work around I've added 0.00001 to each of them whenever screen rotation is detected.

Regards,
JC

**Comment by maxkonovalov** [comment]
Hi @joaofcosta-pt! Did you try manually triggering `setNeedsDisplay` or similar functions?

**Comment by CustomCraftDev** [comment]
this seems to work nicely without any ugly animations:

```
override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
  super.viewWillTransition(to: size, with: coordinator)
  self.collectionView.collectionViewLayout.invalidateLayout()
        
  coordinator.animate(alongsideTransition: { (_) in
    self.collectionView.visibleCells.forEach { (cell) in
      if let c = (cell as? CompareCell)?.progressGroup {
        c.ring1.ringProgressLayer.display()
        c.ring2.ringProgressLayer.display()
        c.ring3.ringProgressLayer.display()
      }
    }
  }, completion: nil )
}

**Comment by joaofcosta-pt** [comment]
Hi there,
 You're right - everything running smoothly. It was an issue that I happened when I've changed to xcode 10 and using the latest version of MKRingProgressView: I had to delete all my animation code and rewrite it again, as the stodyboard editor started to crash everythime I used it.

Sorry for not reporting anything - I've just came back from some holidays and only when I received this comment I realized I have't posted my conclusions here.
